torch/_library/custom_ops.py

```diff
@@ -942,7 +945,8 @@ def _maybe_get_opdef(
         return op
     if isinstance(op, _ops.OpOverload):
         op = op._name
-    assert isinstance(op, str)
+    if not isinstance(op, str):
```

Comment:
Nit: many of these should be TypeError.

Reply:
I agree, but to avoid any BC-breaking implication of this migration, I'm leaving that as a follow up.